From f224ea59a0bff02d8826b5d1dc1701c920c2498b Mon Sep 17 00:00:00 2001
From: bengris32 <bengris32@protonmail.ch>
Date: Fri, 15 Dec 2023 22:05:40 +0000
Subject: [PATCH] ImsService: Export receivers that aren't exclusively
 registered for system broadcasts

* Remove incompatible androidprv:tooltipVerticalPadding while here.

Signed-off-by: bengris32 <bengris32@protonmail.ch>
---
 AndroidManifest.xml                                       | 2 +-
 smali/com/mediatek/ims/ImsConfigManager.smali             | 8 ++++----
 smali/com/mediatek/ims/MtkSuppServExt.smali               | 6 ++++--
 .../ims/config/internal/ImsConfigController.smali         | 6 ++++--
 smali/com/mediatek/ims/internal/ImsVTProviderUtil.smali   | 6 ++++--
 .../mediatek/ims/rcs/MtkRcsCapabilityExchangeImpl.smali   | 6 ++++--
 smali/com/mediatek/wfo/impl/ImsLocationHandler.smali      | 6 ++++--
 smali/com/mediatek/wfo/impl/MwiService.smali              | 6 ++++--
 smali/com/mediatek/wfo/impl/WfcHandler.smali              | 6 ++++--
 smali/com/mediatek/wfo/impl/WifiPdnHandler.smali          | 6 ++++--
 10 files changed, 37 insertions(+), 21 deletions(-)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index a94f6a1..bde8445 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -62,7 +62,7 @@
                 <action android:name="android.telephony.ims.ImsService"/>
             </intent-filter>
         </service>
-        <receiver android:exported="false" android:name="com.mediatek.ims.ImsKeeperReceiver" androidprv:tooltipVerticalPadding="true">
+        <receiver android:exported="false" android:name="com.mediatek.ims.ImsKeeperReceiver">
             <intent-filter>
                 <action android:name="android.telephony.action.CARRIER_CONFIG_CHANGED"/>
             </intent-filter>
diff --git a/smali/com/mediatek/ims/ImsConfigManager.smali b/smali/com/mediatek/ims/ImsConfigManager.smali
index 45dac22..a35553d 100644
--- a/smali/com/mediatek/ims/ImsConfigManager.smali
+++ b/smali/com/mediatek/ims/ImsConfigManager.smali
@@ -129,7 +129,7 @@
 .end method
 
 .method public constructor <init>(Landroid/content/Context;[Lcom/mediatek/ims/ril/ImsCommandsInterface;)V
-    .locals 3
+    .locals 4
     .param p1, "context"    # Landroid/content/Context;
     .param p2, "imsRILAdapters"    # [Lcom/mediatek/ims/ril/ImsCommandsInterface;
 
@@ -206,10 +206,10 @@
 
     invoke-virtual {v1, v2}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
 
-    .line 155
-    invoke-virtual {p1, v0, v1}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
-
     .line 157
+    const/4 v3, 0x2
+
+    invoke-virtual {p1, v0, v1, v3}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
     return-void
 .end method
 
diff --git a/smali/com/mediatek/ims/MtkSuppServExt.smali b/smali/com/mediatek/ims/MtkSuppServExt.smali
index 2bdb945..aeca6ce 100644
--- a/smali/com/mediatek/ims/MtkSuppServExt.smali
+++ b/smali/com/mediatek/ims/MtkSuppServExt.smali
@@ -1644,7 +1644,7 @@
 .end method
 
 .method private registerBroadcastReceiver()V
-    .locals 3
+    .locals 4
 
     .line 465
     new-instance v0, Landroid/content/IntentFilter;
@@ -1677,7 +1677,9 @@
 
     iget-object v2, p0, Lcom/mediatek/ims/MtkSuppServExt;->mBroadcastReceiver:Landroid/content/BroadcastReceiver;
 
-    invoke-virtual {v1, v2, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v3, 0x2
+
+    invoke-virtual {v1, v2, v0, v3}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 472
     iget-object v1, p0, Lcom/mediatek/ims/MtkSuppServExt;->mContext:Landroid/content/Context;
diff --git a/smali/com/mediatek/ims/config/internal/ImsConfigController.smali b/smali/com/mediatek/ims/config/internal/ImsConfigController.smali
index c635d7a..81990ce 100644
--- a/smali/com/mediatek/ims/config/internal/ImsConfigController.smali
+++ b/smali/com/mediatek/ims/config/internal/ImsConfigController.smali
@@ -232,7 +232,7 @@
 .end method
 
 .method public constructor <init>(Landroid/content/Context;ILcom/mediatek/ims/ril/ImsCommandsInterface;)V
-    .locals 7
+    .locals 8
     .param p1, "context"    # Landroid/content/Context;
     .param p2, "phoneId"    # I
     .param p3, "imsRilAdapter"    # Lcom/mediatek/ims/ril/ImsCommandsInterface;
@@ -482,7 +482,9 @@
 
     iget-object v5, p0, Lcom/mediatek/ims/config/internal/ImsConfigController;->mReceiver:Landroid/content/BroadcastReceiver;
 
-    invoke-virtual {v4, v5, v3}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v7, 0x2
+
+    invoke-virtual {v4, v5, v3, v7}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 155
     iget-object v4, p0, Lcom/mediatek/ims/config/internal/ImsConfigController;->mRilAdapter:Lcom/mediatek/ims/ril/ImsCommandsInterface;
diff --git a/smali/com/mediatek/ims/internal/ImsVTProviderUtil.smali b/smali/com/mediatek/ims/internal/ImsVTProviderUtil.smali
index a02edf4..b25a13a 100644
--- a/smali/com/mediatek/ims/internal/ImsVTProviderUtil.smali
+++ b/smali/com/mediatek/ims/internal/ImsVTProviderUtil.smali
@@ -2384,7 +2384,7 @@
 .end method
 
 .method public setContextAndInitRefVTPInternal(Landroid/content/Context;)V
-    .locals 4
+    .locals 5
     .param p1, "context"    # Landroid/content/Context;
 
     .line 1069
@@ -2467,7 +2467,9 @@
 
     iget-object v3, p0, Lcom/mediatek/ims/internal/ImsVTProviderUtil;->mFeatureValueReceiver:Lcom/mediatek/ims/internal/ImsVTProviderUtil$FeatureValueReceiver;
 
-    invoke-virtual {v2, v3, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v4, 0x2
+
+    invoke-virtual {v2, v3, v0, v4}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 1087
     .end local v0    # "filter":Landroid/content/IntentFilter;
diff --git a/smali/com/mediatek/ims/rcs/MtkRcsCapabilityExchangeImpl.smali b/smali/com/mediatek/ims/rcs/MtkRcsCapabilityExchangeImpl.smali
index 0d5eb9b..e447cd8 100644
--- a/smali/com/mediatek/ims/rcs/MtkRcsCapabilityExchangeImpl.smali
+++ b/smali/com/mediatek/ims/rcs/MtkRcsCapabilityExchangeImpl.smali
@@ -84,7 +84,7 @@
 .end method
 
 .method public constructor <init>(ILandroid/content/Context;Landroid/telephony/ims/stub/CapabilityExchangeEventListener;)V
-    .locals 5
+    .locals 6
     .param p1, "slotId"    # I
     .param p2, "context"    # Landroid/content/Context;
     .param p3, "listener"    # Landroid/telephony/ims/stub/CapabilityExchangeEventListener;
@@ -246,7 +246,9 @@
 
     invoke-direct {v3, v4}, Landroid/content/IntentFilter;-><init>(Ljava/lang/String;)V
 
-    invoke-virtual {v1, v2, v3}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v4, 0x2
+
+    invoke-virtual {v1, v2, v3, v4}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     goto :goto_0
 
diff --git a/smali/com/mediatek/wfo/impl/ImsLocationHandler.smali b/smali/com/mediatek/wfo/impl/ImsLocationHandler.smali
index d943888..0cf6e5d 100644
--- a/smali/com/mediatek/wfo/impl/ImsLocationHandler.smali
+++ b/smali/com/mediatek/wfo/impl/ImsLocationHandler.smali
@@ -5175,7 +5175,7 @@
 .end method
 
 .method private registerForBroadcast()V
-    .locals 3
+    .locals 4
 
     .line 1562
     new-instance v0, Landroid/content/IntentFilter;
@@ -5208,7 +5208,9 @@
 
     iget-object v2, p0, Lcom/mediatek/wfo/impl/ImsLocationHandler;->mReceiver:Landroid/content/BroadcastReceiver;
 
-    invoke-virtual {v1, v2, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v3, 0x2
+
+    invoke-virtual {v1, v2, v0, v3}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 1569
     return-void
diff --git a/smali/com/mediatek/wfo/impl/MwiService.smali b/smali/com/mediatek/wfo/impl/MwiService.smali
index 41aaa15..f6682b3 100644
--- a/smali/com/mediatek/wfo/impl/MwiService.smali
+++ b/smali/com/mediatek/wfo/impl/MwiService.smali
@@ -132,7 +132,7 @@
 .end method
 
 .method public constructor <init>(Landroid/content/Context;)V
-    .locals 8
+    .locals 9
     .param p1, "context"    # Landroid/content/Context;
 
     .line 90
@@ -348,10 +348,12 @@
 
     invoke-virtual {v2, v3}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
 
+    const/4 v8, 0x2
+
     .line 123
     iget-object v3, p0, Lcom/mediatek/wfo/impl/MwiService;->mMultiSimConfigChangedReceiver:Landroid/content/BroadcastReceiver;
 
-    invoke-virtual {p1, v3, v2}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    invoke-virtual {p1, v3, v2, v8}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 125
     .end local v2    # "multiSimConfigChanged":Landroid/content/IntentFilter;
diff --git a/smali/com/mediatek/wfo/impl/WfcHandler.smali b/smali/com/mediatek/wfo/impl/WfcHandler.smali
index 6232884..c12879a 100644
--- a/smali/com/mediatek/wfo/impl/WfcHandler.smali
+++ b/smali/com/mediatek/wfo/impl/WfcHandler.smali
@@ -3898,7 +3898,7 @@
 .end method
 
 .method private registerForBroadcast()V
-    .locals 3
+    .locals 4
 
     .line 654
     new-instance v0, Landroid/content/IntentFilter;
@@ -3951,7 +3951,9 @@
 
     iget-object v2, p0, Lcom/mediatek/wfo/impl/WfcHandler;->mReceiver:Landroid/content/BroadcastReceiver;
 
-    invoke-virtual {v1, v2, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v3, 0x2
+
+    invoke-virtual {v1, v2, v0, v3}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 668
     return-void
diff --git a/smali/com/mediatek/wfo/impl/WifiPdnHandler.smali b/smali/com/mediatek/wfo/impl/WifiPdnHandler.smali
index e806430..be9f140 100644
--- a/smali/com/mediatek/wfo/impl/WifiPdnHandler.smali
+++ b/smali/com/mediatek/wfo/impl/WifiPdnHandler.smali
@@ -4132,7 +4132,7 @@
 .end method
 
 .method private registerForBroadcast()V
-    .locals 3
+    .locals 4
 
     .line 851
     new-instance v0, Landroid/content/IntentFilter;
@@ -4170,7 +4170,9 @@
 
     iget-object v2, p0, Lcom/mediatek/wfo/impl/WifiPdnHandler;->mReceiver:Landroid/content/BroadcastReceiver;
 
-    invoke-virtual {v1, v2, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+    const/4 v3, 0x2
+
+    invoke-virtual {v1, v2, v0, v3 }, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;I)Landroid/content/Intent;
 
     .line 858
     return-void
-- 
2.53.0

